### DISCUSSION TITLE
Better Tailwind Support

### DISCUSSION THREAD

**Discussion by Pukimaa** [withastro/starlight#1046] [Feature Requests]
### What version of `starlight` are you using?

0.12.0

### What is your idea?

My idea is to replace a bunch of variables with tailwind variables, so you can just replace the color in the tailwind config, similiar to the neutral colors and the primary/accent color. I mostly want this to easily change the colors of the aside notes.

### Why is this feature necessary?

This would be extremely helpful to incorperate the branding of the project without having to dig through the css files of starlight and writing CSS manually.

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Thanks for the feedback.

As Tailwind support is optional in Starlight, replacing some variables by Tailwind variables is definitely not something we want to do directly in Starlight core.

Regarding the specific of the question about customizing note aside colors, Starlight now uses [cascade layers](https://starlight.astro.build/guides/css-and-tailwind/#cascade-layers) like Tailwind v4, which means you can easily override Starlight's styles without having to care about specificity, as any custom unlayered CSS will override the default Starlight styles.

What this means in practice is that you can just add CSS targeting the aside note class `.starlight-aside--note` in the existing `src/styles/global.css` file of your Starlight project and it'll work as expected. In such CSS, you can also use any Tailwind CSS variables, e.g. `var(--color-blue-500)`:

```css
.starlight-aside--note {
  background-color: var(--color-blue-500);
}
```
